diff/diff.go

```diff
@@ -1,10 +1,64 @@
 package diff
 
-import "bytes"
+import (
+	"bytes"
+	"time"
+)
 
-// NOTE: types are code-generated in diff.pb.go.
+// A FileDiff represents a unified diff for a single file.
+//
+// A file unified diff has a header that resembles the following:
+//
+//   --- oldname	2009-10-11 15:12:20.000000000 -0700
+//   +++ newname	2009-10-11 15:12:30.000000000 -0700
+type FileDiff struct {
+	// the original name of the file
+	OrigName string
+	// the original timestamp (nil if not present)
+	OrigTime *time.Time
+	// the new name of the file (often same as OrigName)
+	NewName string
+	// the new timestamp (nil if not present)
+	NewTime *time.Time
+	// extended header lines (e.g., git's "new mode <mode>", "rename from <path>", etc.)
+	Extended []string
+	// hunks that were changed from orig to new
+	Hunks []*Hunk
```

Comment:
Why a slice of pointers? `[]Hunk` should work too.

Reply:
Made it as it was in generated file. Maybe better to left it as making `[]Hunk` requires changes in every entry where used `(*HunksReader) ReadAllHunks()`, since the function returns `[]*Hunk`?

Reply:
ah, right. Yeah, better leave it then to avoid breaking the public API.